### PR TITLE
expression: implement vectorized evaluation for 'builtinRandSig'

### PR DESCRIPTION
**Pull request by tsthght**
### What problem does this PR solve? 
Implement vectorized evaluation for builtinRandSig, for #12105

### What is changed and how it works?
```
BenchmarkVectorizedBuiltinMathFunc/builtinRandSig-VecBuiltinFunc-8         	  300000	      5953 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinMathFunc/builtinRandSig-NonVecBuiltinFunc-8      	   50000	     29702 ns/op	       0 B/op	       0 allocs/op
```
### Check List
Returns random numbers, so cannot write test cases

### CODE REVIEW

**Comment by tsthght** [comment]
> Please do a benchmark for this function and post the result in the description. @tsthght
> For correct testing, let us discuss in this [PR/12656](https://github.com/pingcap/tidb/pull/12656).

ok

**Comment by XuHuaiyu** [comment]
We can add the unit test in `TestRand` function.

**Comment by SunRunAway** [comment]
Please consider write a separated test.

**Comment by tsthght** [comment]
> Please consider write a separated test.

I had add a separated testcase:TestVectorizedBuiltinMathFuncForRand. I don't know whether it's right. PTAL @SunRunAway

File: expression/bench_test.go [hunk 1]
```diff
@@ -931,6 +931,73 @@ func testVectorizedBuiltinFunc(c *C, vecExprCases vecExprBenchCases) {
 	}
 }
 
+// testVectorizedBuiltinFuncForRand is used to verify that the vectorized
+// expression is evaluated correctly
+func testVectorizedBuiltinFuncForRand(c *C, vecExprCases vecExprBenchCases) {
```
Comment (qw4990) [inline expression/bench_test.go]: This test seems too complicated.
How about keeping it as simple as possible like:
```
randFunc := ConstructRandFunction()
input := ConstructInput()
var result Column
randFunc.VecEvalReal(input, result)
for i := range result {
    assert (0 <= result[i] < 1)
}
```

**Pull request merged**